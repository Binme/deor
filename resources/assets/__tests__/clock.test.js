import Vue from 'vue/dist/vue.js';
import timemachine from 'timemachine';
import Clock from './../js/components/Clock.vue';

describe('vue clock component', () => {
    Vue.component('clock', Clock);

    beforeEach(() => {
        timemachine.config({
            dateString: 'December 25, 1991 13:12:59'
        });

        document.body.innerHTML = `
            <div id="app">
                <p>Hello World!</p>
                <clock></clock>
            </div>
        `;
    });

    it('clock shows the current time', async () => {
        await createVm();

        expect(document.body.innerHTML).toMatchSnapshot();
    });
});

async function createVm() {
    const vm = new Vue({
        el: '#app',
    });

    await Vue.nextTick();

    return vm.$children[0];
}
