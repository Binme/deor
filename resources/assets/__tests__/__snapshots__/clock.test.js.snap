// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vue clock component clock shows the current time 1`] = `
"
            <div id=\\"app\\"><p>Hello World!</p> <div class=\\"clock wrapper\\"><div class=\\"month-year\\">
      December 1991
    </div> <div class=\\"date-time\\">
      Wed 01:12:59 PM
    </div></div></div>
        "
`;
